### ISSUE TITLE
Migrate JDT core related wiki pages

### ISSUE THREAD

**Issue by trancexpress** [eclipse-jdt/eclipse.jdt.core#753]
See: https://github.com/eclipse-jdt/eclipse.jdt/issues/54

We only want to move JDT core wiki pages at first, and this at: https://github.com/eclipse-jdt/eclipse.jdt.core.wiki.git

**Comment by trancexpress** [comment]
Links will be a big problem. Also the wiki doesn't show images, even though opening an `.md` file with `Okular`does show images... Likely we have to give a full path link to the image...

Current state is at: https://github.com/trancexpress/eclipse.jdt.core/wiki/

**Comment by trancexpress** [comment]
Looks like there are pages not seen in the JDT category. E.g. I found this one:

https://wiki.eclipse.org/JDT_Core_Programmer_Guide/ECJ/Lambda

**Comment by trancexpress** [comment]
Still have to fix this one: https://github.com/trancexpress/eclipse.jdt.core/wiki/Java

There are a lot of `div` tags that don't work anymore.

**Comment by trancexpress** [comment]
@iloveeclipse I think I fixed the biggest problems, we can check the contents next and prune outdated information. And fix other problems that I overlooked.

**Comment by trancexpress** [comment]
I did what I could for the committer FAQ page: https://github.com/trancexpress/eclipse.jdt.core/wiki/JDT-Core-Committer-FAQ

But there is a lot about release management. I don't know how the processes has changed with the move to GitHub. Or if the process still exists in general. @iloveeclipse you'll have to handle that part, either by finding someone who knows how the process works... or we just delete that part of the page / make a reference to the JDT Eclipse wiki that will eventually be deleted.

**Comment by trancexpress** [comment]
I fin a lot of outdated information here: https://github.com/trancexpress/eclipse.jdt.core/wiki/JDT-Core-Index-Programmer-Guide

I've added a warning and referenced the bugs that disabled and later on removed the new index. We'll have to prune the information about the new index... if nothing is left of the page after that we delete it.

**Comment by trancexpress** [comment]
OK, I went over the pages, fixed formatting and so on. I've added disclaimers (at the start) for pages where there is outdated information.

For the grouping/categories, I don't think there is an option. I found this feature request: https://github.com/community/community/discussions/23914

**Comment by iloveeclipse** [comment]
No worries, it should be straightforward with git. Will do next week.

**Comment by trancexpress** [comment]
I've rebased my `master` branch on top of the JDT one. So updating the JDT core wiki should be simple now.

**Comment by trancexpress** [comment]
@iloveeclipse I've fixed links in: https://github.com/trancexpress/eclipse.jdt.core/wiki/ECJ

Anything else here? The committer process is done now, I can push the wiki pages myself. Should I?

**Comment by iloveeclipse** [comment]
Check the wery first link in 
https://github.com/trancexpress/eclipse.jdt.core/wiki/ECJ-Lambda

Could it be, that inter-wiki links are all broken?

**Comment by iloveeclipse** [comment]
Same here, see links to JDT Core Programmer Guide/Completion
https://github.com/trancexpress/eclipse.jdt.core/wiki/ECJ-Parse

**Comment by iloveeclipse** [comment]
> None of the links work out of the box after migration

Wow, this is a major issue, have you reported it to the transition bug? Or is that JDT specific?

**Comment by iloveeclipse** [comment]
... and yes, if you can, just push to JDT wiki. There is no reason to work on private fork anymore, all the fine tuning can be done in the official repo.

**Comment by trancexpress** [comment]
> Wow, this is a major issue, have you reported it to the transition bug? Or is that JDT specific?

I think its an issue with the migration script (https://github.com/outofcontrol/mediawiki-to-gfm). Since its not related to Eclipse I assumed it makes no sense to open a bug anywhere. Its probably a special case of how GitHub wiki links work. Probably the generated `.md` links work fine for just markdown files. But not for the GitHub wiki pages. 



> If that is the case, shouldn't there be some visible "errors" generatedcin html, or "todos" in code?

There is `"wikilink"` text next to each link. But the link can point into nothing.



> ... and yes, if you can, just push to JDT wiki. There is no reason to work on private fork anymore, all the fine tuning can be done in the official repo.

OK I'll fix the issues you found and I'll push.

**Comment by iloveeclipse** [comment]
**Please** report the link issue to the migration bug, to share our pain with others, so they are warned and may be someone could find / share a fix.

**Comment by trancexpress** [comment]
OK, I commented here: https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/681#note_1086886

Wiki pages are pushed now, see: https://github.com/eclipse-jdt/eclipse.jdt.core/wiki

**Comment by iloveeclipse** [comment]
@stephan-herrmann : regarding structure: I believe it can help if you clone the wiki repo locally to check where the **actual** wiki files are. Clone https://github.com/eclipse-jdt/eclipse.jdt.core.wiki.git and import the project.

![image](https://user-images.githubusercontent.com/964108/221860288-766ee314-ff1d-4584-8c52-7b4f6a79ac2f.png)

**Comment by trancexpress** [comment]
@stephan-herrmann , for adding links please check the existing ones, e.g. to link a page and a subpage:

```
[JDT Core Programmer Guide/ECJ/Completion\#Lambda Specifics](ECJ-Completion#Lambda-Specifics "wikilink")
```

Relative paths to the links don't work for GitHub, or at least I couldn't get them to work. This is painful when using a markdown editor, since the links can't be tested in the editor...

The hierarchy of pages is gone yes, best I could do was to add prefixes to pages. E.g., `ECJ-AST`.

Also some of the nicer html formattings from the Eclipse wiki didn't make it through the migration. E.g. some color tags still exist but GitHub doesn't show colors.

**Comment by stephan-herrmann** [comment]
... "painful" describes it well. I thought github wiki was mature software, but a preview speaking a different language from the saved version is ... ("embarrassing"? ...? ).

**Comment by trancexpress** [comment]
> I thought github wiki was mature software

During the migration, to me at least the wiki functionality looked more like an afterthought. Overall GitHub doesn't seem to be the best platform for big projects like Eclipse, but not much to do about it at this point.

**Comment by iloveeclipse** [comment]
> embarrassing

Why? It delivers minimum viable product to get people locked & money from organizations that follow the hype. That's how the IT world works since decades. Nothing really embarrassing. Also remember who is the current github owner is and which track of record it has delivering buggy software to customers :-)

**Comment by trancexpress** [comment]
Anything else to discuss here? If not, closing the issue.

**Comment by iloveeclipse** [comment]
No, I think if we find a new issue we can fix in the wiki itself.
Thanks you Simeon!
